c/core.go

```diff
@@ -108,19 +117,27 @@ func (cs ChildSpec) Name() string {
 // waitTimeout is the internal function used by Child to wait for the execution
 // of it's thread to stop.
 func waitTimeout(
-	terminateCh chan struct{},
+	terminateCh <-chan ChildNotification,
 ) func(Shutdown) error {
 	return func(shutdown Shutdown) error {
 		switch shutdown.tag {
 		case infinityT:
 			// We wait forever for the result
-			<-terminateCh
-			return nil
+			notification, ok := <-terminateCh
+			if !ok {
+				return nil
+			}
+			// A child may have terminated with an error
+			return notification.Unwrap()
 		case timeoutT:
 			// we wait until some duration
 			select {
-			case <-terminateCh:
-				return nil
+			case notification, ok := <-terminateCh:
+				if !ok {
+					return nil
+				}
+				// A child may have terminated with an error
+				return notification.Unwrap()
 			case <-time.After(shutdown.duration):
 				return errors.New("Child shutdown timeout")
```

Comment:
It may change from 5s, you can specify the `Shutdown` value of a Client using the `WithShutdown` combinator [[link](https://github.com/capatazlib/go-capataz/blob/90dfe7487184829356da5fb918e7d8ef8922a3f1/c/core.go#L20)]

Reply:
Uhmm, sadly we are not using an error type that can add custom attributes (e.g. `error.New("...").KV("key", val)`), and I think that is overkill for now without an specific use-case.

Reply:
Troubleshooting the logs. I cannot foresee a specific case, but I figured if you are going to bother to log, and timeout values may vary, might as well note what it was.